testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js

```diff
@@ -3893,3 +3894,164 @@ QUnit.module('calendarOptions', moduleConfig, () => {
+
+                dateBox.option('opened', false);
+
+                assert.strictEqual($(this.instance.startDateField()).attr('aria-controls'), $(this.instance.content()).attr('id'), 'aria-controls on startDateInput');
+                assert.strictEqual($(this.instance.endDateField()).attr('aria-controls'), $(this.instance.content()).attr('id'), 'aria-controls on endDateInput');
+            });
+        });
+
+        QUnit.test('aria-activedescendant on input should point to the contoured cell\'s id in calendar', function(assert) {
+            if(devices.real().deviceType !== 'desktop') {
+                assert.ok(true, 'test does not actual for mobile devices');
+                return;
+            }
+
+            this.reinit({
+                calendarOptions: {
+                    currentDate: new Date(2021, 9, 17),
+                },
+                deferRendering,
+            });
+
+            const getContouredCell = () => {
+                return $(this.instance.content()).find(`.${CALENDAR_CONTOURED_CELL_CLASS}`);
+            };
+
+            let contouredCellID = getContouredCell().attr('id');
+
+            assert.strictEqual($(this.instance.startDateField()).attr('aria-activedescendant'), deferRendering ? undefined : contouredCellID, 'aria-activedescendant on startDateInput');
+            assert.strictEqual($(this.instance.endDateField()).attr('aria-activedescendant'), deferRendering ? undefined : contouredCellID, 'aria-activedescendant on endDateInput');
+
+            this.instance.option('opened', true);
+
+            contouredCellID = getContouredCell().attr('id');
+
+            assert.strictEqual($(this.instance.startDateField()).attr('aria-activedescendant'), contouredCellID, 'aria-activedescendant on startDateInput');
+            assert.strictEqual($(this.instance.endDateField()).attr('aria-activedescendant'), contouredCellID, 'aria-activedescendant on endDateInput');
+
+            const keyboard = keyboardMock($(this.instance.startDateField()));
+
+            keyboard.keyDown('right');
+
+            const newContouredCellID = getContouredCell().attr('id');
+
+            assert.strictEqual(newContouredCellID === contouredCellID, false, 'countoured cell is changed');
+            assert.strictEqual($(this.instance.startDateField()).attr('aria-activedescendant'), newContouredCellID, 'aria-activedescendant on startDateInput');
+            assert.strictEqual($(this.instance.endDateField()).attr('aria-activedescendant'), newContouredCellID, 'aria-activedescendant on endDateInput');
+
+            this.instance.option('opened', false);
+
+            assert.strictEqual($(this.instance.startDateField()).attr('aria-activedescendant'), newContouredCellID, 'aria-activedescendant on startDateInput');
+            assert.strictEqual($(this.instance.endDateField()).attr('aria-activedescendant'), newContouredCellID, 'aria-activedescendant on endDateInput');
```

Comment:
Too many asserts. I think we can split this test

Reply:
https://github.com/DevExpress/DevExtreme/pull/24798/commits/fadda309c6c15b8d9459ef9e5c474605c0ef1a54